src/applications/vaos/newAppointmentFlow.js

```diff
@@ -41,19 +51,36 @@ export default {
   },
   typeOfCare: {
     url: '/new-appointment',
-    async next(state) {
+    async next(state, dispatch) {
+      const newAppointment = getNewAppointment(state);
       let nextState = 'vaFacility';
 
       if (isSleepCare(state)) {
         nextState = 'typeOfSleepCare';
       } else if (isCommunityCare(state)) {
         try {
-          const data = await getCommunityCare(
-            '/vaos/community-care/eligibility',
+          // Check if user registered systems support comminity care...
+          let systems = newAppointment.systems;
+          if (!systems) {
+            systems = (await dispatch(getUserSystems())).systems;
```

Comment:
I don't really love this, but it's either this, or pass `getState()` to `next()` as well.

Reply:
Do we need the system details for anything? It looks like we just need the system ids, which is the number after `dfn-` that `getUserSystems` parses out.

I wonder if it would be cleaner to just do `getSystemIdentifiers` in here and not try to get it from the Redux state or update the Redux state afterward. If we want to avoid making unnecessary api requests, we could do what I did for `getPastAppointments` and memoize `getSystemIdentifiers`. It's unlikely that we'll need to refresh the systems someone is registered in over the course of a session.

Reply:
Good call.  I've memoized `getSystemIdentifiers()`.  Let me know if that's what you had in mind.